src/main/java/com/timgroup/statsd/StatsDProcessor.java

```diff
@@ -147,11 +147,11 @@ public void run() {
     }
 
     boolean isShutdown() {
-        executor.shutdown();
         return shutdown;
     }
 
     void shutdown() {
+        executor.shutdown();
         shutdown = true;
```

Comment:
`StatsdSender.shutdown` seems to set the boolean before initiating the shutdown (https://github.com/DataDog/java-dogstatsd-client/blob/1dd87626d49cf2d45dc9249d62b59fa3f51f3d2f/src/main/java/com/timgroup/statsd/StatsDSender.java#L164-L165), it might make more sense here as well to reflect that the shutdown has started, since `executor.shutdown` doesn't guarantee upon return that processing has entirely stopped.

Also, might make sense to make the boolean `volatile`? (not sure of potential perf impact)

Reply:
`shutdown` is already volatile.

Reply:
you're right, not sure how I missed that 😄  thanks!